modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -526,6 +526,12 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self._modin_frame._index_cache is None:
+            new_self = Map.register(
+                lambda df, *args, **kwargs: df.reset_index(**kwargs)
+            )(self, **kwargs)
```

Comment:
We have to apply this function across the whole column axis, right? `new_self = self._modin_frame.apply_full_axis(axis=0, lambda df: df.reset_index(**kwargs)` Otherwise we'll have an incorrect index in partitions:
```
  a b
1 . .
2 . .    <- part1
------
3 . .    <- part2
4 . .

If we use 'Map' to reset indices:
  a b
0 . .
1 . .    <- part1
------
0 . .    <- part2
1 . .

When it should be:
  a b
0 . .
1 . .    <- part1
------
2 . .    <- part2
3 . .
```

Reply:
You are right, but it will become significantly slower.